src/web-push-lib.js

```diff
@@ -174,6 +176,14 @@ WebPushLib.prototype.generateRequestDetails = function(subscription, payload, op
           console.warn('Attempt to use proxy option, but invalid type it should be a string or proxy options object.');
         }
       }
+
+      if (options.agent) {
+        if (options.agent instanceof https.Agent) {
+          agent = options.agent;
+        } else {
+          console.warn('Attempt to use agent option, but invalid type it should be an instanceof https.Agent.');
+        }
+      }
     }
 
```

Comment:
Log a warning if both proxy and agent are set (saying that agent will be ignored because proxy is set).